cli/build-public-library.js

```diff
@@ -120,62 +105,51 @@ function processFiles(skyPagesConfig) {
   );
 }
 
-/**
- * Creates a UMD JavaScript bundle.
- * @param {*} skyPagesConfig
- * @param {*} webpack
- */
-function createBundle(skyPagesConfig, webpack) {
-  const webpackConfig = require('../config/webpack/build-public-library.webpack.config');
-  const config = webpackConfig.getWebpackConfig(skyPagesConfig);
-  return runCompiler(webpack, config);
-}
-
 /**
  * Transpiles TypeScript files into JavaScript files
  * to be included with the NPM package.
  */
 function transpile() {
-  return new Promise((resolve, reject) => {
-    const result = spawn.sync(
-      skyPagesConfigUtil.spaPath('node_modules', '.bin', 'ngc'),
-      [
-        '--project',
-        skyPagesConfigUtil.spaPathTemp('tsconfig.json')
-      ],
-      { stdio: 'inherit' }
-    );
-
-    // Catch ngc errors.
-    if (result.err) {
-      reject(result.err);
-      return;
-    }
+  const projectConfigPath = skyPagesConfigUtil.spaPathTemp('ng-package.json');
+  const tsConfigPath = skyPagesConfigUtil.spaPathTemp('tsconfig.json');
+
+  return ngPackage.ngPackagr()
+    .forProject(projectConfigPath)
+    .withTsConfig(tsConfigPath)
+    .build();
+}
 
-    // Catch non-zero status codes.
-    if (result.status !== 0) {
-      reject(new Error(`Angular compiler (ngc) exited with status code ${result.status}.`));
-      return;
+function writePackagerConfig() {
+  const ngPackageConfig = {
+    lib: {
+      entryFile: 'index.ts'
```

Comment:
@Blackbaud-BobbyEarl @Blackbaud-PaulCrowder What do you think about adding the `whitelistedNonPeerDependencies` option here? It would throw errors if certain packages were listed in the package.json `dependencies`. Might be useful to lint what our consumers list as `dependencies`.
https://github.com/ng-packagr/ng-packagr/blob/v5/docs/dependencies.md#whitelisting-the-dependencies-section

Reply:
That sounds like a good idea to me.